Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -86,11 +102,89 @@ endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId merchan
+updateDriverDailyZscore ride rideDate driverZscore currDriverDailyStat = do
+  dailyLeaderBoardConfig <- Esq.runInReplica $ QLeaderConfig.findLeaderBoardConfigbyType LConfig.DAILY >>= fromMaybeM (InternalError "Leaderboard configs not present")
+  let lbExpiry = getSeconds dailyLeaderBoardConfig.leaderBoardExpiry
+  case driverZscore of
+    Nothing -> do
+      let currDriverZscore = currDriverDailyStat.dailyTotalRides * dailyZscoreBase + (getMeters currDriverDailyStat.dailyTotalDistance)
+      Hedis.zAddExp (makeDailyDriverLeaderBoardKey rideDate) (getId ride.driverId) (fromIntegral currDriverZscore) $ lbExpiry * dailyLeaderBoardConfig.numberOfSets
+      pure ()
+    Just zscore -> do
+      let (prevTotalRides, prevTotalDistance) = getRidesAndDistancefromZscore zscore dailyZscoreBase
+      let currTotaRides = prevTotalRides + 1
+      let currTotalDist = prevTotalDistance + currDriverDailyStat.dailyTotalDistance
+      let currZscore = currTotaRides * dailyZscoreBase + getMeters currTotalDist
+      Hedis.zAddExp (makeDailyDriverLeaderBoardKey rideDate) (getId ride.driverId) (fromIntegral currZscore) $ lbExpiry * dailyLeaderBoardConfig.numberOfSets
+      pure ()
+
+updateDriverWeeklyZscore :: (Esq.EsqDBReplicaFlow m r, Metrics.CoreMetrics m, CacheFlow m r, Hedis.HedisFlow m r, MonadFlow m) => Ride.Ride -> Day -> Day -> Maybe Double -> (Id DP.Person, Maybe Int, Maybe Meters) -> m ()
+updateDriverWeeklyZscore ride weekStartDate weekEndDate driverZscore (_, mbWeeklyTotalRides, mbWeeklyTotalDistance) = do
+  weeklyLeaderBoardConfig <- Esq.runInReplica $ QLeaderConfig.findLeaderBoardConfigbyType LConfig.WEEKLY >>= fromMaybeM (InternalError "Leaderboard configs not present")
+  let lbExpiry = getSeconds weeklyLeaderBoardConfig.leaderBoardExpiry
+  case driverZscore of
+    Nothing -> do
+      let currDriverZscore = (fromMaybe 1 mbWeeklyTotalRides) * weeklyZscoreBase + getMeters (fromMaybe 0 mbWeeklyTotalDistance)
+      Hedis.zAddExp (makeWeeklyDriverLeaderBoardKey weekStartDate weekEndDate) (getId ride.driverId) (fromIntegral currDriverZscore) $ lbExpiry * weeklyLeaderBoardConfig.numberOfSets
+      pure ()
+    Just zscore -> do
+      let (prevTotalRides, prevTotalDistance) = getRidesAndDistancefromZscore zscore weeklyZscoreBase
+      let currTotaRides = prevTotalRides + 1
+      let currTotalDist = prevTotalDistance + (fromMaybe 0 ride.chargeableDistance)
+      let currZscore = currTotaRides * weeklyZscoreBase + getMeters currTotalDist
+      Hedis.zAddExp (makeWeeklyDriverLeaderBoardKey weekStartDate weekEndDate) (getId ride.driverId) (fromIntegral currZscore) $ lbExpiry * weeklyLeaderBoardConfig.numberOfSets
+      pure ()
+
+dailyZscoreBase :: Int
+dailyZscoreBase = 100000000
+
+weeklyZscoreBase :: Int
+weeklyZscoreBase = 1000000000
+
+makeDailyDriverLeaderBoardKey :: Day -> Text
+makeDailyDriverLeaderBoardKey rideDate =
+  "DailyDriverLeaderBoardKey " <> show rideDate
```

Comment:
can you keep `:` instead of `<space>` ? we are following these convention for all redis keys across the app.